src/software/ai/hl/stp/tactic/goalie/goalie_tactic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "software/ai/hl/stp/tactic/goalie/goalie_tactic.h"
+
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/action/stop_action.h"  // TODO (#1888): remove this dependency
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/point.h"
+
+GoalieTactic::GoalieTactic(std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config)
+    : Tactic(true,
+             {RobotCapability::Move, RobotCapability::Dribble, RobotCapability::Chip}),
+      fsm(),
+      goalie_tactic_config(goalie_tactic_config),
+      control_params{
+          GoalieFSM::ControlParams{.goalie_tactic_config = goalie_tactic_config}}
+{
+}
+
+void GoalieTactic::updateWorldParams(const World &world) {}
+
+void GoalieTactic::updateControlParams(
+    std::shared_ptr<const GoalieTacticConfig> updated_goalie_tactic_config)
+{
+    // Update the control parameters stored by this Tactic
+    control_params.goalie_tactic_config = updated_goalie_tactic_config;
+}
+
+double GoalieTactic::calculateRobotCost(const Robot &robot, const World &world) const
+{
+    // We don't prefer any particular robot to be the goalie, as there should only
+    // ever be one robot that can act as the goalie
+    return 0.5;
+}
+
+void GoalieTactic::calculateNextAction(ActionCoroutine::push_type &yield)
+{
+    auto stop_action = std::make_shared<StopAction>(false);
+
+    do
+    {
+        stop_action->updateControlParams(*robot_, false);
+        yield(stop_action);
```

Comment:
There used to be 3 cases in goalie_tactic: 1) Panic and stop the ball, 2) Chip it out, 3) block the shot. I was wondering where that logic has been migrated to?

Reply:
All the logic has been moved to goalie_fsm.h, with slightly updated states: panic_s, dribble_s, chip_s, and position_to_block_s

Reply:
Ah I see, thanks and good job on this PR!